File: src/main/java/com/chatchatabc/jpademojava/domain/service/PostService.java
package com.chatchatabc.jpademojava.domain.service;

import com.chatchatabc.jpademojava.domain.model.Post;
import org.springframework.stereotype.Service;

@Service
public interface PostService {

    /**
     * Create post
     *
     * @param userId
     * @param post
     * @return
     */
    Post create(String userId, Post post) throws Exception;

    /**
     * Update posts
     *
     * @param postId
     * @param newPostInfo
     * @return
     */
    Post update(String postId, Post newPostInfo) throws Exception;

    /**
     * Delete post
     *
     * @param postId
     */
    void delete(String postId) throws Exception;
}

File: src/main/kotlin/com/chatchatabc/jpademo/domain/service/PostService.kt
package com.chatchatabc.jpademo.domain.service

import com.chatchatabc.jpademo.domain.model.Post
import org.springframework.stereotype.Service

@Service
interface PostService {

    /**
     * Create Post
     */
    fun create(userId:String, post: Post): Post

    /**
     * Update Post
     */
    fun update(postId: String, newPostInfo: Post): Post

    /**
     * Delete Post
     */
    fun delete(postId: String)
}